src/common/containers/with-repository.js

```diff
@@ -22,7 +23,11 @@ function withRepository(WrappedComponent) {
     render() {
       const { fullName, ...passThroughProps } = this.props; // eslint-disable-line no-unused-vars
 
-      return (this.props.repository
+      // only render if current repo in the store matches repo in URL params
+      const { repository } = this.props;
+      const isReady = repository && repository.fullName === fullName;
+
+      return (isReady
```

Comment:
Won't this hide the `WrappedComponent` on occasions that the repository is being fetched? What if a child component one day wants to show a spinner while repo data is being loaded?